lib/core/cardano-wallet-core.cabal

```diff
@@ -419,6 +421,9 @@ test-suite unit
     , wai
     , wai-extra
     , warp
+  if (flag(scrypt))
+    cpp-options: -DHAVE_SCRYPT
+    build-depends: scrypt
```

Comment:
Ah, yes. The unit tests were still unconditionally dependent on `scrypt`, unlike the library, so this seems good. Haven't checked whether it passes, but this seems fair nonetheless.